File: CE303/StockMarketC#/Client/BackgroundInputThread.cs
using System;
using System.Threading;

namespace Client 
{
    class BackgroundInputThread
    {
        private readonly Client client;
        private readonly int id;

        public BackgroundInputThread(Client client)
        {
            this.client = client;
            this.id = client.Id;
        }

        public void Run()
        {
            try
            {
                //Run loop to keep Thread alive
                while(true)
                {
                    string line = Console.ReadLine().Trim().ToLower();
                    switch(line)
                    {
                        case "quit":
                            client.Dispose();
                            Environment.Exit(0);
                            break;
                        case "id":
                            Console.WriteLine($"Your ID : {id}");
                            break;
                        case "rules":
                            Program.PrintRules("wait");
                            break;
                        case "update":
                            Program.PrintTraders(client, "wait");
                            break;
                        case "continue":
                            if(client.GetStockAmount() == 1)
                                return;
                            else
                                Console.WriteLine("You cannot yet proceed. Please wait...");
                            break;
                        default:
                            Console.WriteLine("Unknown command");
                            break;
                    }
                }
            }
            catch(ThreadInterruptedException)
            {
                return;
            }
        }
    }
}

File: CE303/StockMarketJava/src/Client/BackgroundInputThread.java
package Client;

import java.util.Scanner;

public class BackgroundInputThread implements Runnable {

    private final Client client;
    private final int id;

    public BackgroundInputThread(Client client){
        this.client = client;
        this.id = client.getId();
    }

    @Override
    public void run() {
        Scanner read = new Scanner(System.in);
        //Run loop to keep Thread alive
        while (true) {
            //Inner loop to check if there is the line to read input
            while (read.hasNextLine()) {
                String line = read.nextLine().trim().toLowerCase();
                switch (line) {
                    case "quit":
                        client.close();
                        System.exit(0);
                        break;
                    case "id":
                        System.out.println("Your ID: " + id);
                        break;
                    case "rules":
                        ClientProgram.printRules("wait");
                        break;
                    case "update":
                        ClientProgram.printTraders(client, "wait");
                        break;
                    case "continue":
                        if(client.getStockAmount() == 1)
                            return;
                        else
                            System.out.println("You cannot yet proceed. Please wait...");
                        break;
                    default:
                        System.out.println("Unknown command");
                }
                if(Thread.currentThread().isInterrupted())
                    return;
            }
            if(Thread.currentThread().isInterrupted())
                return;
        }
    }
}